integration/test/Test/LegalHold.hs

```diff
@@ -906,15 +906,33 @@ testLHDisableBeforeApproval = do
       >>= assertStatus 200
     getBob'sStatus `shouldMatch` "disabled"
 
+-- ---------
+-- WPB-10783
+-- ---------
+testBlockLHForMLSUsers :: (HasCallStack) => App ()
+testBlockLHForMLSUsers = do
+  -- scenario 1:
+  -- if charlie is in any MLS conversation, he cannot approve to be put under legalhold
+  (charlie, tid, []) <- createTeam OwnDomain 1
+  [charlie1] <- traverse (createMLSClient def) [charlie]
+  void $ createNewGroup charlie1
+  void $ createAddCommit charlie1 [charlie] >>= sendAndConsumeCommitBundle
+
+  legalholdWhitelistTeam tid charlie >>= assertStatus 200
+  withMockServer def lhMockApp \lhDomAndPort _chan -> do
+    postLegalHoldSettings tid charlie (mkLegalHoldSettings lhDomAndPort) >>= assertStatus 201
+    requestLegalHoldDevice tid charlie charlie `bindResponse` do
+      assertLabel 409 "mls-legal-hold-not-allowed"
```

Comment:
This is nice ~ denying the request seems like the better option.

Reply:
```suggestion
    requestLegalHoldDevice tid charlie charlie `bindResponse`
      assertLabel 409 "mls-legal-hold-not-allowed"
```

Reply:
Format becomes very ugly without the do.